libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotExternalEventsHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core.Handlers
+{
+    public class BotExternalEventsHandler : BotMessageHandlerBase
```

Comment:
>BotExternalEventsHandler [](start = 17, length = 24)

can we add comment which describes the function of this class?

Reply:
So I'd actually love to talk about this more. I actually think the handlers should be marked `internal` because they are not really something we want developers to ever use themselves. They're just _how_ we do the handling of the messages and they really shouldn't ever take a direct dependency on that IMNSHO. Aside from it being a good idea from a pure API surface area perspective, it also means we wouldn't need to apply the full set of documentation comments like we need to on public classes and I would, at most, just need to put some simple developer level comments on here for maintainers.

The only reason they are public right now was because we hadn't set up the proper assembly trust between this assembly and its test assembly. I had done it at one point, but there was some discussion with @cleemullins that it wasn't worth it and I should just make things public. This is a good reason not to just make things public.